client/dashboard/sites/settings-transfer-site/invitation-email-sent.tsx

```diff
@@ -0,0 +1,93 @@
+import { useMutation } from '@tanstack/react-query';
+import { useRouter } from '@tanstack/react-router';
+import { ExternalLink, Spinner } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { createInterpolateElement } from '@wordpress/element';
+import { sprintf, __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useEffect, useState } from 'react';
+import { siteOwnerTransferConfirmMutation } from '../../app/queries';
+import Notice from '../../components/notice';
+import type { SiteTransferConfirmation } from '../../data/types';
+
+export function InvitationEmailSent( {
+	siteSlug,
+	confirmationHash,
+}: {
+	siteSlug: string;
+	confirmationHash: string;
+} ) {
+	const [ newOwnerEmail, setNewOwnerEmail ] = useState( '' );
+	const [ hasError, setHasError ] = useState( false );
+	const mutation = useMutation( siteOwnerTransferConfirmMutation( siteSlug ) );
+	const { createSuccessNotice } = useDispatch( noticesStore );
+	const router = useRouter();
+
+	// The page is accessed via the confirmation email, so this is the only place where the request can be triggered.
+	// It would be better if
+	// * we have the server trigger the mutation before rendering the page
+	// * we first show a page, clarifying what is going to happen with an "accept" button or something.
+	useEffect( () => {
+		mutation.mutate(
+			{ hash: confirmationHash },
+			{
+				onSuccess: ( { transfer, new_owner_email }: SiteTransferConfirmation ) => {
+					if ( transfer ) {
+						createSuccessNotice(
+							sprintf(
+								/* translators: %(newOwnerEmail)s - the new owner's email */
+								__( 'The site has been successfully transferred to %(newOwnerEmail)s.' ),
+								{
+									newOwnerEmail: new_owner_email,
+								}
+							),
+							{ type: 'snackbar' }
+						);
+						router.navigate( { to: '/sites', replace: true } );
+						return;
+					}
+
+					setNewOwnerEmail( new_owner_email );
```

Comment:
Seems like this line is not really necessary? We don't show the new owner email anymore after the transfer is successful.